### PR TITLE
gerogettica/add help command

### PR DESCRIPTION
**Pull request by georgettica**
- fix args issue
- go mod tidy
- add usage

### CODE REVIEW

**Comment by georgettica** [comment]
@dastergon  here you go :)

**Comment by georgettica** [comment]
Disregard the GitHub actions, these will be removed from the PR
/hold
In case you lgtm

**Comment by iamkirkbater** [comment]
I know you're saying to specifically disregard the GH actions but we can't use GH actions on the openshift org :(

**Comment by georgettica** [comment]
That is why I have it on my local repo

Forgot I had this when I made the PR

**Comment by georgettica** [comment]
@iamkirkbater fixed :)

File: cmd/servicelog/list.go [hunk 1]
```diff
@@ -19,8 +19,12 @@ var listCmd = &cobra.Command{
 	Use:   "list",
```
Comment (iamkirkbater) [inline cmd/servicelog/list.go:19]:
```suggestion
	Use:   "list [flags] [options] cluster-identifier",
```

File: cmd/servicelog/list.go [hunk 2]
```diff
@@ -19,8 +19,12 @@ var listCmd = &cobra.Command{
...
 	Short: "gets all servicelog messages for a given cluster",
 	// validate only clusterid is provided
-	Args: cobra.ExactArgs(1),
+	Args: cobra.ArbitraryArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			cmd.Help()
+			return fmt.Errorf("the argument count is not equal to 1, it is %d", len(args))
```
Comment (iamkirkbater) [inline cmd/servicelog/list.go:26]: Can we change this error message to something a bit more descriptive?

Something like `cluster-identifier was not provided. Please provide a cluster ID, UUID, or name for this command`?

The difference is how the output will read:

```
$ ./osdctl servicelog list
gets all servicelog messages for a given cluster

Options:
  -A, --all-messages=false: Toggle if we should see all of the messages or only SRE-P specific ones

Usage:
  osdctl servicelog list [flags] [options] cluster-id

Use "osdctl options" for a list of global command-line options (applies to all commands).
Error: the argument count is not equal to 1, it is 0
the argument count is not equal to 1, it is 0
```

vs

```
$ ./osdctl servicelog list
gets all servicelog messages for a given cluster

Options:
  -A, --all-messages=false: Toggle if we should see all of the messages or only SRE-P specific ones

Usage:
  osdctl servicelog list [flags] [options] cluster-id

Use "osdctl options" for a list of global command-line options (applies to all commands).
Error: No Cluster Identifier provided.
No Cluster Identifier provided.
```

File: cmd/servicelog/list.go (hunk 1 above)
Comment (iamkirkbater) [reply cmd/servicelog/list.go:19]: This change will change the help output from

```
$ osdctl servicelog list --help
gets all servicelog messages for a given cluster

Options:
  -A, --all-messages=false: Toggle if we should see all of the messages or only SRE-P specific ones

Usage:
  osdctl servicelog list [flags] [options]

Use "osdctl options" for a list of global command-line options (applies to all commands).
```

To
```
$ ./osdctl servicelog list --help
gets all servicelog messages for a given cluster

Options:
  -A, --all-messages=false: Toggle if we should see all of the messages or only SRE-P specific ones

Usage:
  osdctl servicelog list [flags] [options] cluster-id

Use "osdctl options" for a list of global command-line options (applies to all commands).
``` 

(note the "Usage" field

**Review by iamkirkbater — CHANGES_REQUESTED** [review]
Few small nits but then this will be solid :D

**Comment by georgettica** [comment]
/unhold

I will squash via UI when the lgtm is given

**Review by iamkirkbater — APPROVED** [review]
/lgtm
/unhold

**Comment by iamkirkbater** [comment]
Nice addition with "SilenceErrors" so it doesn't show the error twice.

**Pull request merged**